server/workflow/workflow_server.go

```diff
@@ -299,9 +299,6 @@ func (s *workflowServer) DeleteWorkflow(ctx context.Context, req *workflowpkg.Wo
 	if err != nil {
 		return nil, err
 	}
-	if wf.Finalizers != nil && !req.Force {
```

Comment:
Will this change have unintended side effects? For example, if the workflow has a non-GC related finalizer, should we still require --force for the delete command?